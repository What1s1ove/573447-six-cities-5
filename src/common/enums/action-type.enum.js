const PlacesDataActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
};

export {PlacesDataActionType};
