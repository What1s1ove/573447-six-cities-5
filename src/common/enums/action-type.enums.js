const PlacesDataActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
  UPDATE_OFFER: `UPDATE_OFFER`,
};

const PlaceActionType = {
  LOAD_OFFER: `LOAD_OFFER`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  LOAD_SIMILAR_OFFERS: `LOAD_SIMILAR_OFFERS`,
  UPDATE_SIMILAR_OFFER: `UPDATE_SIMILAR_OFFER`,
};

const FavoritesActionType = {
  LOAD_FAVORITES: `LOAD_FAVORITES`,
  UPDATE_FAVORITE: `UPDATE_FAVORITE`,
};

const UserActionType = {
  SET_AUTH_STATUS: `SET_AUTH_STATUS`,
  SET_USER: `SET_USER`,
  REDIRECT_TO_ROUTE: `REDIRECT_TO_ROUTE`,
};

export {
  PlacesDataActionType,
  PlaceActionType,
  UserActionType,
  FavoritesActionType,
};
