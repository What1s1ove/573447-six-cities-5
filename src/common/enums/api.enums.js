const HttpCode = {
  SUCCESS: 200,
  UNAUTHORIZED: 401,
};

const ApiRoute = {
  LOGIN: `/login`,
  HOTELS: `/hotels`,
  FAVORITE: `/favorite`,
  COMMENTS: `/comments`,
};

export {HttpCode, ApiRoute};
