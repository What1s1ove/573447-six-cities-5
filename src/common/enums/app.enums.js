const AppRoute = {
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
  MAIN: `/`,
  OFFER: `/offer`,
  OFFER_ID: `/offer/:id`,
};

const AppConfig = {
  OFFERS_COUNT: 40,
  REVIEWS_COUNT: 5,
};

const SortType = {
  POPULAR: `Popular`,
  TO_HIGHT: `Price: low to high`,
  TO_LOW: `Price: high to low`,
  TOP_RATE: `Top rated first`,
};

export {AppRoute, AppConfig, SortType};
