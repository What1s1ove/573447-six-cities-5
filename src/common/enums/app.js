const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  OFFER_ID: `/offer/:id`,
};

export {AppRoute};
