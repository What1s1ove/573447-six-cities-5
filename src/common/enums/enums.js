export * from './app.enums';
export * from './offers.enums';
export * from './api.enums';
export * from './action-type.enum';
