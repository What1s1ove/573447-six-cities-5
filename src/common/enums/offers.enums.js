const OfferCity = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

const OfferRoomType = {
  APARTMENT: `apartment`,
  HOTEL: `hotel`,
  HOUSE: `house`,
  ROOM: `room`,
};

const OfferFavoriteStatus = {
  TRUE: 1,
  FALSE: 0,
};

export {OfferCity, OfferRoomType, OfferFavoriteStatus};
