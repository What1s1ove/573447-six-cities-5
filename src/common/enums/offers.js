const OfferCity = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`,
};

const OfferRoomType = {
  APARTMENT: `Apartment`,
  ROOM: `Room`,
  HOUSE: `House`,
  HOTEL: `Hotel`,
};

export {OfferCity, OfferRoomType};
