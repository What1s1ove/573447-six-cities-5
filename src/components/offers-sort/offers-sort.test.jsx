import * as React from 'react';
import {render} from 'enzyme';
import {SortType} from '~/common/enums/enums';
import OffersSort from './offers-sort';

describe(`OffersSort component`, () => {
  it(`should render OffersSort component`, () => {
    const component = render(
        <OffersSort
          activeSort={SortType.POPULAR}
          sorts={Object.values(SortType)}
          onSortChange={jest.fn()}
        />
    );

    expect(component).toMatchSnapshot();
  });
});
