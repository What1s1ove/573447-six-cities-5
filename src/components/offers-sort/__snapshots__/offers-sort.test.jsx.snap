// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersSort component should render OffersSort component 1`] = `
<form
  action="#"
  class="places__sorting"
  method="get"
>
  <span
    class="places__sorting-caption"
  >
    Sort by
  </span>
  <select
    class="places__sorting-type"
  >
    <option
      class="places__option"
      selected=""
      value="Popular"
    >
      Popular
    </option>
    <option
      class="places__option"
      value="Price: low to high"
    >
      Price: low to high
    </option>
    <option
      class="places__option"
      value="Price: high to low"
    >
      Price: high to low
    </option>
    <option
      class="places__option"
      value="Top rated first"
    >
      Top rated first
    </option>
  </select>
</form>
`;
